### PR TITLE
Change address to location and make it exclusive to specialists

### LINKED ISSUE #67: Rename address to location and make the field exclusive to specialists
Currently the `Address` field is in the `Person` superclass, but only `Specialist` should have an `Address`. With the addition of the `Age` field to `Patient` which would use the `a/` prefix, it is better we rename `Address` to `Location` and change its prefix to `l/` so that there is less confusion for users.

### PR DESCRIPTION
**Pull request by alvinlim277**
Make `Location` exclusive to `Specialists` and change the syntax from a/ to l/ so as not to be confused with the `Age` syntax (to be added later)
- Add some additional test cases for EditSpecialistDescriptor and EditPatientDescriptor
- Fix small bug with `AddCommand`, the whole list of `Persons` would be shown after execution instead of the specific personType.
  - The constructor of `AddCommand` could be further changed to take the personType directly as well, but that is minor and can be decided on in the future.

Resolves #67

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 1]
```diff
@@ -155,7 +155,8 @@ public ObservableList<Person> getFilteredPersonList() {
 
         @Override
         public void updateFilteredPersonList(Predicate<Person> predicate) {
-            throw new AssertionError("This method should not be called.");
```
Comment (vansh284) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:158]: Quoting the header comment, this is "a default model stub that has all of the methods failing". I'm not sure if this should be done. 
If I'm not wrong, subclasses of this class need to override a method if it wants to allow that method to be called. In this case, I believe the `updateFilteredPersonList` method can be overridden in the `ModelStubAcceptingPersonAdded` class to allow the tests to work as intended (not 100% sure about this).

**Review by vansh284 — COMMENTED** [review]
Good job on code quality and rigorous testing. Seems like just a couple of minor errors to resolve before the code can be merged.

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java (hunk 1 above)
Comment (alvinlim277) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:158]: I think what you are suggesting is more correct regardless, will make the changes 👍

**Review by vansh284 — APPROVED** [review]
LGTM

**Pull request merged**